macos/FluentUI/Notification/Notification.swift

```diff
@@ -0,0 +1,346 @@
+					}).buttonStyle(PlainButtonStyle())
+				}
+			}
+		}
+
+		let messageButtonAction = state.messageButtonAction
+		@ViewBuilder
+		var innerContents: some View {
+			if hasCenteredText {
+				HStack {
+					Spacer()
+					textContainer
+					Spacer()
+				}
+				.frame(minHeight: Constants.minimumHeight)
+			} else {
+				let horizontalSpacing = Constants.horizontalSpacing
+				HStack(spacing: 0) {
+					HStack(spacing: horizontalSpacing) {
+						image
+						textContainer
+						Spacer(minLength: 0)
+					}
+					.accessibilityElement(children: .combine)
+					button
+						.layoutPriority(1)
+				}
+				.frame(minHeight: Constants.minimumHeight)
+				.padding(.horizontal, Constants.horizontalPadding)
+				.clipped()
+			}
+		}
+
+		@ViewBuilder
+		var notification: some View {
+			innerContents
+				.background(
+					backgroundColor
+					.clipShape(RoundedRectangle(cornerRadius: state.style.isToast ? Constants.toastCornerRadius : Constants.barCornerRadius))
+				)
+				.overlay(Rectangle().frame(width: nil, height: state.style == .primaryOutlineBar ? Constants.outlineWidth : 0, alignment: .top).foregroundColor(Constants.neutralBackgroundColor), alignment: .top)
+				.onTapGesture {
+					if let messageAction = messageButtonAction {
+						messageAction()
+					}
+				}
+		}
+
+		return notification
+	}
+
+	var state: MSFNotificationStateImpl
+
+	var backgroundColor: Color
+
+	var foregroundColor: Color
+
+	private var hasImage: Bool {
+		state.style.isToast && state.image != nil
+	}
+
+	private var hasSecondTextRow: Bool {
+		guard state.title != nil else {
+			return false
+		}
+
+		return state.style.isToast
+	}
+
+	private var hasCenteredText: Bool {
+		!state.style.isToast && state.actionButtonAction == nil
+	}
+
+	private struct Constants {
+		static let primaryBackgroundColor: Color = Color(FluentUI.Colors.primaryTint30)
+		static let primaryForegroundColor: Color = Color(FluentUI.Colors.primaryShade20)
+		static let neutralBackgroundColor: Color = Color(FluentUI.Colors.Palette.gray100.color)
+		static let neutralForegroundColor: Color = Color(FluentUI.Colors.Palette.gray900.color)
+		static let primaryOutlineBarBackgroundColor: Color = Color(.white)
+		static let primaryOutlineBarForegroundColor: Color = Color(FluentUI.Colors.primary)
+		static let dangerToastBackgroundColor: Color = Color(FluentUI.Colors.Palette.dangerTint30.color)
+		static let dangerToastForegroundColor: Color = Color(FluentUI.Colors.Palette.dangerShade20.color)
+		static let warningToastBackgroundColor: Color = Color(FluentUI.Colors.Palette.warningTint30.color)
+		static let warningToastForegroundColor: Color = Color(FluentUI.Colors.Palette.warningShade30.color)
+
+		static let toastCornerRadius: CGFloat = 12
+		static let barCornerRadius: CGFloat = 0
+		static let horizontalPadding: CGFloat = 16
+		static let verticalPadding: CGFloat = 12
+		static let horizontalSpacing: CGFloat = 16
+		static let minimumHeight: CGFloat = 52
```

Comment:
because we don't have redlines for this i was curious if we could follow similar sizes of existing apps 
![image](https://user-images.githubusercontent.com/20715435/219783105-e7637155-6b4d-4f0a-9c23-a978e5e36dd5.png)